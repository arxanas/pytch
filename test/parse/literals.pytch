let foo = 1
let bar = "bar"
let baz = [1, 2]

let qux = ()
let quux = (3,)  # one-tuple
let grault = (3)  # parenthesized int literal
let xyzzy = (3, 4)
