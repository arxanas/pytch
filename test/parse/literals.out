(Program
  (LetPlainStmt
    (Ident 'foo')
    (IntLiteral 1))
  (LetPlainStmt
    (Ident 'bar')
    (StringLiteral 'bar'))
  (LetPlainStmt
    (Ident 'baz')
    (ListLiteral
      (IntLiteral 1)
      (IntLiteral 2)))
  (LetPlainStmt
    (Ident 'qux')
    (TupleLiteral))
  (LetPlainStmt
    (Ident 'quux')
    (TupleLiteral
      (IntLiteral 3)))
  (LetPlainStmt
    (Ident 'grault')
    (IntLiteral 3))
  (LetPlainStmt
    (Ident 'xyzzy')
    (TupleLiteral
      (IntLiteral 3)
      (IntLiteral 4))))
